optimism/types.go

```diff
@@ -300,6 +305,46 @@ type GraphQL interface {
 	Query(ctx context.Context, input string) (string, error)
 }
 
+// L2Receipt represents the results of a transaction.
+type L2Receipt struct {
```

Comment:
I originally tried doing that but it ended up becoming a bigger change than I'd like. 2 things is the case
1. It appears the [receipt type in op-geth](https://github.com/ethereum-optimism/op-geth/blob/optimism/core/types/receipt.go#L55-L95) is incorrect. It's using a simple bigInt instead of some type that works with hex strings.
2. Importing the op-geth Receipt type may require a rewrite of other modules that use Receipt. Bumping this [replace](https://github.com/Inphi/optimism-rosetta/blob/master/go.mod#L102) affects a lot of other modules which I'm not able to fully test due to the scope of this change

Reply:
Quite. Updating go.mod deps breaks other rosetta dependencies. I couldn't quite figure that out either. But this is OK for the time being.

Though fyi: Receipt has a custom decoder you can use https://github.com/ethereum-optimism/op-geth/blob/optimism/core/types/gen_receipt_json.go#L17 that handles hex.